base/server/src/main/java/com/netscape/cmscore/usrgrp/User.java

```diff
@@ -49,77 +52,77 @@ public class User implements JSONSerializer {
     /**
      * Constant for userScope
      */
-    public final static String ATTR_SCOPE = "userScope";
+    public static final String ATTR_SCOPE = "userScope";
 
     /**
      * Constant for userName
      */
-    public final static String ATTR_NAME = "userName";
+    public static final String ATTR_NAME = "userName";
 
     /**
      * Constant for userId
      */
-    public final static String ATTR_ID = "userId";
+    public static final String ATTR_ID = "userId";
 
     /**
      * Constant for userFullName
      */
-    public final static String ATTR_FULLNAME = "userFullName";
+    public static final String ATTR_FULLNAME = "userFullName";
 
     /**
      * Constant for userPassword
      */
-    public final static String ATTR_PASSWORD = "userPassword";
+    public static final String ATTR_PASSWORD = "userPassword";
 
     /**
      * Constant for userState
      */
-    public final static String ATTR_STATE = "userstate";
+    public static final String ATTR_STATE = "userstate";
 
     /**
      * Constant for userEmail
      */
-    public final static String ATTR_EMAIL = "userEmail";
+    public static final String ATTR_EMAIL = "userEmail";
 
     /**
      * Constant for usertype
      */
-    public final static String ATTR_USERTYPE = "usertype";
+    public static final String ATTR_USERTYPE = "usertype";
 
     /**
      * Constant for usertype
      */
-    public final static String ATTR_TPS_PROFILES = "tpsProfiles";
+    public static final String ATTR_TPS_PROFILES = "tpsProfiles";
 
-    public final static String ATTR_X509_CERTIFICATES = "userCertificates";
-    public final static String ATTRIBUTES = "attributes";
+    public static final String ATTR_X509_CERTIFICATES = "userCertificates";
+    public static final String ATTRIBUTES = "attributes";
+    public static final String CMS_BASE_INVALID_ATTRIBUTE = "CMS_BASE_INVALID_ATTRIBUTE";
 
-    private String mUserid = null;
-    private String mUserDN = null;
-    private String mFullName = null;
-    private String mPassword = null;
-    private String mEmail = null;
-    private String mPhone = null;
-    private String mState = null;
-    private String mCertDN = null;
+    private String userid = null;
+    private String userDN = null;
+    private String fullName = null;
+    private String password = null;
+    private String email = null;
+    private String phone = null;
+    private String state = null;
+    private String certDN = null;
     private String mUserType = null;
-    private X509Certificate mx509Certs[] = null;
-    private List<String> tpsProfiles = null;
-    private List<LDAPAttribute> mAttributes = null;
+    private X509Certificate[] x509Certs;
+    private List<String> tpsProfiles;
+    private List<LDAPAttribute> attributes;
 
-    private static final Vector<String> mNames = new Vector<>();
+    private static final Set<String> mNames = new HashSet<>();
     static {
```

Comment:
Same as above.

Reply:
This one needs instantiating with a concrete implementation as `Set` is an abstract interface.

Reply:
That is is OK but it could be `names` instead of `mNames`

Reply:
Turns out there is a new way in Java 9, TIL.

Reply:
> That is is OK but it could be `names` instead of `mNames`

Oh yeah :laughing: - sorry, fixed.

Reply:
> Turns out there is a new way in Java 9, TIL.

It looks much better!